pkg/server/registry/apigroups/acorn/secrets/storage.go

```diff
@@ -12,11 +13,16 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func NewStorage(c kclient.WithWatch) rest.Storage {
+func NewStorage(c kclient.WithWatch, middlewares ...middleware.CompleteStrategy) rest.Storage {
 	translated := translation.NewTranslationStrategy(&Translator{
 		c: c,
 	}, remote.NewRemote(&corev1.Secret{}, c))
 	remoteResource := publicname.NewStrategy(translated)
+
+	for i := len(middlewares) - 1; i >= 0; i-- {
+		remoteResource = middlewares[i](remoteResource)
+	}
```

Comment:
Since you moved the type to its own package, I'd like to see an `ApplyMiddlewares` function there that does this.

Reply:
Right on, I'll add that.

Reply:
Added. Thoughts?